cmake/libutils.cmake

```diff
@@ -196,18 +197,29 @@ MACRO(MERGE_STATIC_LIBS TARGET OUTPUT_NAME LIBS_TO_MERGE)
       )  
     ELSE()
       # Generic Unix, Cygwin or MinGW. In post-build step, call
-      # script, that extracts objects from archives with "ar x" 
-      # and repacks them with "ar r"
+      # script, that uses a MRI script to append static archives.
+      SET(MRI_SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.mri")
```

Comment:
I'm not really sure what are you doing here. Can you explain?


Otherwise, I read from CMake documentation for FILE(GENERATE OUTPUT) https://cmake.org/cmake/help/latest/command/file.html

OUTPUT <output-file>
Specify the output file name to generate. Use generator expressions such as $<CONFIG> to specify a configuration-specific output file name. Multiple configurations may generate the same output file only if the generated content is identical. Otherwise, the <output-file> must evaluate to an unique name for each configuration. A relative path (after evaluating generator expressions) is treated with respect to the value of CMAKE_CURRENT_BINARY_DIR. See policy CMP0070.

Did you try that with a multiconfig generator? I have not tried anything like that apart from VS, and Xcode, but nowadays there is even a  Ninja Multi-Config , so maybe it is worth trying. I'm not insisting mostly curious.

Reply:
The libraries at the time the macro is called is still a generator expression:
```
/build-mariadb-server-10.5-mysql_release$ more libmysqld/mysqlserver.mri.tpl
CREATE $<TARGET_FILE:mysqlserver>
ADDLIB $<TARGET_FILE:dbug>
ADDLIB $<TARGET_FILE:strings>
ADDLIB $<TARGET_FILE:mysys>
ADDLIB $<TARGET_FILE:mysys_ssl>
ADDLIB $<TARGET_FILE:vio>
```
So this is really only expanded before it calls the POST_BUILD.
I normally have an out of tree build each with their own config. Didn't even try ninja, bad me. I'll need to read up on this.